.github/workflows/gpu-tests.yaml

```diff
@@ -0,0 +1,60 @@
+# This workflow runs all of the very-long tests within main.py
+
+name: Weekly Tests
+
+on:
+  # Runs every Sunday from 7AM UTC
+  schedule:
+    - cron:  '00 7 * * 6'
+  # Allows us to manually start workflow for testing
+  workflow_dispatch:
+
+jobs:
+  build-gem5:
+    runs-on: [self-hosted, linux, x64, build]
+    container: gcr.io/gem5-test/gcn-gpu:latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Scheduled workflows run on the default branch by default. We
+          # therefore need to explicitly checkout the develop branch.
+          ref: develop
+      - name: Build gem5
+        run: scons build/GCN3_X86/gem5.opt -j $(nproc) --ignore-style
+      - uses: actions/upload-artifact@v3
+        with:
+          name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-gcn3
+          path: build/GCN3_X86/gem5.opt
+          retention-days: 5
+      - run: echo "This job's status is ${{ job.status }}."
+
+  HACC-tests:
+    runs-on: [self-hosted, linux, x64, build]
+    container: gcr.io/gem5-test/gcn-gpu:latest
+    needs: build-gem5
+    timeout-minutes: 4320 # 3 day
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Scheduled workflows run on the default branch by default. We
+          # therefore need to explicitly checkout the develop branch.
+          ref: develop
+      - uses: actions/download-artifact@v3
+        with:
+          name: weekly-test-${{ github.run_number }}-attempt-${{ github.run_attempt }}-gem5-build-gcn3
+          path: build/GCN3_X86
+      - run: chmod u+x build/GCN3_X86/gem5.opt
+      - name: make hip directory
+        run: mkdir hip
+      - name: Compile m5ops and x86
+        working-directory: ${{ github.workspace }}/util/m5
+        run: |
+          export TERM=xterm-256color
+          scons build/x86/out/m5
+      - name: Download tests
+        working-directory: ${{ github.workspace }}/hip
+        run: wget http://dist.gem5.org/dist/v22-1/test-progs/halo-finder/ForceTreeTest
```

Comment:
Inspired by this I created this: https://github.com/gem5/gem5/pull/317.

If we merge #317 we can replace this `wget` with `./build/GCN3_X86/gem5.opt util/obtain-resource.py force-tree-test -o ForceTreeTest`. This means if we update ForceTreeTest we can just change the entry in our database rather than update this URL. It'll also mean we always get the right version of this resources for the version of gem5 being tested.

Reply:
Should we wait for #317 then?  As it is, v22-1 is already out of date, however I do not know if there is a more recent binary, so that may be the newest

Reply:
@mkjost0 : This has been added as a resource with the ID `hacc-force-tree`. #317 has been merged so please make this change.